cmd/csi-attacher/main.go

```diff
@@ -120,7 +120,9 @@ func main() {
 			os.Exit(1)
 		}
 
-		supportsService, err := csiConn.SupportsPluginControllerService(ctx)
+		ctxController, cancelController := context.WithTimeout(context.Background(), csiTimeout)
```

Comment:
I don't like 3 different contexts. If you think that 1 second is too short for GetDriverName+SupportsPluginControllerService+SupportsControllerPublish together, then we could increase the context timeout to 3 seconds instead of using three different ones.

Reply:
After increasing Probe timeout, I still meet timeout error in external attacher container. We may not call waitForDriverReady during  GetDriverName+SupportsPluginControllerService+SupportsControllerPublish.

```
# kubectl logs csi-neonsan-controller-0 csi-attacher
I0827 07:58:09.721682       1 main.go:75] Version: v0.2.0-35-g5a4cc3b-dirty
I0827 07:58:09.727437       1 connection.go:88] Connecting to /var/lib/csi/sockets/pluginproxy/csi.sock
I0827 07:58:09.727923       1 connection.go:115] Still trying, connection is CONNECTING
I0827 07:58:09.728326       1 connection.go:112] Connected
I0827 07:58:09.728351       1 connection.go:235] GRPC call: /csi.v0.Identity/GetPluginInfo
I0827 07:58:09.728363       1 connection.go:236] GRPC request: 
I0827 07:58:09.731362       1 connection.go:238] GRPC response: name:"csi-neonsan" vendor_version:"0.3.0" 
I0827 07:58:09.731484       1 connection.go:239] GRPC error: <nil>
I0827 07:58:09.731495       1 main.go:115] CSI driver name: "csi-neonsan"
I0827 07:58:09.731585       1 connection.go:235] GRPC call: /csi.v0.Identity/Probe
I0827 07:58:09.731629       1 connection.go:236] GRPC request: 
I0827 07:58:11.783301       1 connection.go:238] GRPC response: ready:<value:true > 
I0827 07:58:11.783424       1 connection.go:239] GRPC error: <nil>
I0827 07:58:11.783439       1 main.go:202] Probe succeeded
I0827 07:58:11.783453       1 connection.go:235] GRPC call: /csi.v0.Identity/GetPluginCapabilities
I0827 07:58:11.783464       1 connection.go:236] GRPC request: 
I0827 07:58:11.784170       1 connection.go:238] GRPC response: 
I0827 07:58:11.784301       1 connection.go:239] GRPC error: rpc error: code = DeadlineExceeded desc = context deadline exceeded
E0827 07:58:11.784501       1 main.go:125] rpc error: code = DeadlineExceeded desc = context deadline exceeded

```